File: ActivitySuggestion/ManuallyReview/PhpClass/IACMContainer.php
<?php

require_once __DIR__ . '/eventContainer.php' ;

class IACMContainer extends EventContainer{
	private $urlPrefix;
	//private $pureText;
	public function __construct($xml){
		parent::__construct($xml);
		$this->urlPrefix = 'http://www.iacm.gov.mo/';
	}
	public function Parse(){
		$this->ParseName();
		$this->ParseDescription();
		$this->ParsePoster();
		$this->ParseStartDate();
		$this->ParseEndDate();
		$this->ParseTel();
		return $this->ToJSON();
		
	}
	private function ParseTel(){
		$pattern = "/查詢電話 :(.{9,9})/";
		$ret = preg_match($pattern, $this->xmlObj->plaintext, $matches);
		if ($ret) {
			$this->tel = $matches[1];
			return $this->tel;
		}
		return null;
	}
	private function ParsePoster(){
		$imgTag = $this->xmlObj->find('img', 0);
		
		if ($imgTag){
			$this->poster = $this->urlPrefix . $imgTag->src;
			return $this->poster;
		}
		return null;
	}
	private function ParseName(){
		$nameTag = $this->xmlObj->find('h1',0);
		if ($nameTag){
			$this->name = $nameTag->innertext;
			return $this->name;
		}
		return null;
	}
	private function ParseDescription(){
		$desNode = $this->xmlObj->find("div[id=contentDetail]", 0);
		if ($desNode){
			$this->des = $desNode->plaintext;
			return $this->des;
		}
		return null;
	}
	private function ParseStartDate(){
		$pattern = "/開始日期 :(.*?)結束日期/";
		$ret = preg_match($pattern, $this->xmlObj->plaintext, $matches);
		if ($ret) {
			$pattern = "#/#";
			$this->startDate = preg_replace($pattern, "-", $matches[1]) . " 00:00";
			return $this->startDate;
		}
		return null;
	}
	private function ParseEndDate(){
		$pattern = "/結束日期 :(.{10,10})/";
		$ret = preg_match($pattern, $this->xmlObj->plaintext, $matches);
		if ($ret) {
			$pattern = "#/#";
			$this->endDate = preg_replace($pattern, "-", $matches[1]) . " 23:59";
			return $this->endDate;
		}
		return null;
	}
}
?>

File: Crawler/src/IACMContainer.java
import java.net.MalformedURLException;
import java.net.URL;
import java.sql.Date;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

import org.apache.commons.lang3.StringEscapeUtils;
import org.htmlcleaner.CleanerProperties;
import org.htmlcleaner.HtmlCleaner;
import org.htmlcleaner.PrettyXmlSerializer;
import org.htmlcleaner.TagNode;
import org.htmlcleaner.XmlSerializer;

public class IACMContainer extends EventContainer {
	private TagNode tagNode;
	private String urlPrefix;
	private String pureText;
	public IACMContainer(String originalContentXML){
		this.originalContent = originalContentXML;
		this.tagNode = null;
		this.urlPrefix = "http://www.iacm.gov.mo";
	}
	public IACMContainer(TagNode node){
		this.tagNode = node;
		this.urlPrefix = "http://www.iacm.gov.mo";
	}
	public boolean Parse(){
		CleanerProperties props = new CleanerProperties();
		
		// set some properties to non-default values
		props.setTranslateSpecialEntities(true);
		props.setTransResCharsToNCR(true);
		props.setOmitComments(true);

		if (this.tagNode == null){					
			HtmlCleaner hc = new HtmlCleaner(props); 
			this.tagNode = hc.clean(this.originalContent);	
		}
		
		if (this.originalContent.isEmpty()){
			this.originalContent = new PrettyXmlSerializer(props).getAsString(tagNode);
		}
		
		this.pureText = this.tagNode.getText().toString();
		this.ParesName();
		this.ParseDescription();
		this.ParseStartDate();
		this.ParseEndDate();
		this.ParsePoster();
		System.out.println(this.originalContent);
		
		return false;
	}
	private boolean ParsePoster(){
		//System.out.println(this.tagNode.toString());
		TagNode[] hrefs = tagNode.getElementsByName("img", true);
		//XmlSerializer serializer =  new PrettyXmlSerializer(props);
		if (hrefs.length > 0){
			String url = hrefs[0].getAttributeByName("src");
			url = StringEscapeUtils.unescapeXml(url);
			//System.out.println(serializer.getAsString(url));
			System.out.println(urlPrefix + url);
			try {
				this.poster = new URL(urlPrefix + url);
				return true;
			} catch (MalformedURLException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
				return false;
			}
		}
		return false;
	}
	private boolean ParesName(){
		TagNode[] headings = this.tagNode.getElementsByName("h1", true);
		if (headings.length > 0){
			this.name = headings[0].getText().toString();
			return true;
		}
		return false;
	}
	private boolean ParseDescription(){
		TagNode[] desNode = this.tagNode.getElementsByAttValue("id", "contentDetail", true, false);
		if (desNode.length > 0){
			this.des = desNode[0].getText().toString();
			return true;
		}
		return false;
	}
	private boolean ParseHostName(){
		return false;
	}
	private boolean ParseStartDate(){
		Pattern pattern = Pattern.compile("開始日期 :((\n\\s*)*)(.*)(\n\\s*)");
		Matcher matcher = pattern.matcher(this.pureText);
		if (matcher.find()) {
			System.out.println(matcher.group(3));
			this.startDate = matcher.group(3);
			return true;
		}
		return false;
	}
	private boolean ParseEndDate(){
		Pattern pattern = Pattern.compile("結束日期 :((\n\\s*)*)(.*)(\n\\s*)");
		Matcher matcher = pattern.matcher(this.pureText);
		if (matcher.find()) {
			System.out.println(matcher.group(3));
			this.endDate = matcher.group(3);
			return true;
		}
		return false;
	}
}
